### PR TITLE
Revert back - Transactional state store API without retry

### PR DESCRIPTION
**Pull request by youngbupark**
# Description

Revert back - Transactional state store API.

> Note: removed retry for state because of https://github.com/dapr/dapr/issues/1861

## Issue reference

Reverting back to 0.10.0.
@chinzhiweiblank 

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Comment by youngbupark** [comment]
@chinzhiweiblank @yaron2 @amanbha Please take a look at this. I removed state option property. I will create the separate pr for https://github.com/dapr/dapr/issues/1736

**Comment by amanbha** [comment]
IMO, before we merge this, we need to handle this #1736 to surface this api with metadata at transaction level.

**Comment by youngbupark** [comment]
> IMO, before we merge this, we need to handle this #1736 to surface this api with metadata at transaction level.

oh I misunderstand #1736 I thought we need to add one state option(consistency, concurrency) for all requests instead of per state options. Do I misunderstand ? 

cc/ @LMWF

**Comment by LMWF** [comment]
The idea with #1736 is to have metadata at the TransactionalRequest level, instead of using the ones within each .Request field.

**Comment by youngbupark** [comment]
@amanbha @LMWF It requires the change of state request struct in component-contrib first. Changing this struct requires another change in actor state store. I would like to merge this one first. Then I will change component-contrib and create the pr to have metadata and stateoptions on transactional request.

**Comment by amanbha** [comment]
> actor state store. I

@youngbupark If we cant make changes for #1736 in this milestone, then we should keep this PR open and merge in next milestone with all the changes,

**Comment by youngbupark** [comment]
> > actor state store. I
> 
> @youngbupark If we cant make changes for #1736 in this milestone, then we should keep this PR open and merge in next milestone with all the changes, Changes for #1736 can happen in a separate PR, but should be in same milestone.

yes I will take both in the same milestone.

**Comment by amanbha** [comment]
> > > actor state store. I
> > 
> > 
> > @youngbupark If we cant make changes for #1736 in this milestone, then we should keep this PR open and merge in next milestone with all the changes, Changes for #1736 can happen in a separate PR, but should be in same milestone.
> 
> yes I will take both in the same milestone.

Sounds great, thanks!

File: pkg/actors/actors.go [hunk 1]
```diff
@@ -372,7 +372,7 @@ func (a *actorsRuntime) TransactionalStateOperation(ctx context.Context, req *Tr
 	if a.store == nil {
 		return errors.New("actors: state store does not exist or incorrectly configured")
 	}
-	requests := []state.TransactionalRequest{}
+	operations := []state.TransactionalStateOperation{}
```
Comment (youngbupark) [inline pkg/actors/actors.go:375]: @LMWF Do you want to allow users to set metadata for transactional api? it will be breaking change.

**Comment by youngbupark** [comment]
> > > > actor state store. I
> > > 
> > > 
> > > @youngbupark If we cant make changes for #1736 in this milestone, then we should keep this PR open and merge in next milestone with all the changes, Changes for #1736 can happen in a separate PR, but should be in same milestone.
> > 
> > 
> > yes I will take both in the same milestone.
> 
> Sounds great, thanks!

@amanbha @LMWF Please review the pr again. it includes #1736 change.

File: pkg/actors/actors.go (hunk 1 above)
Comment (amanbha) [reply pkg/actors/actors.go:375]: our sdks are not passing it, and for actors I don't foresee passing a state store specific metadata.

Comment (youngbupark) [reply pkg/actors/actors.go:375]: @amanbha Thanks for the confirmation. Once @LMWF reviewed, I will merge it.

**Comment by yaron2** [comment]
Thanks for this contribution, @chinzhiweiblank!

**Pull request merged**